app/src/main/java/com/example/mobiledger/presentation/recordtransaction/AddTransactionDialogFragment.kt

```diff
@@ -52,28 +55,37 @@ class AddTransactionDialogFragment :
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        getCategoryList()
         initToggle()
         initDate()
+        viewModel.getIncomeCategoryList()
+        viewModel.getExpenseCategoryList()
         setUpObserver()
+        getCategoryList(TransactionType.Expense)
     }
 
-    private fun getCategoryList() {
-        categoryList = viewModel.provideCategoryList()
+    private fun getCategoryList(transactionType: TransactionType) {
+        categoryList = if (transactionType == TransactionType.Income)
```

Comment:
update this function if possible

Reply:
Made the list local. For reinitializing the adapter, let it be for now and I will note it down in the doc and will change that in the future.